great_expectations/experimental/checkpoint.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Sequence
+
+from great_expectations.checkpoint import Checkpoint as LegacyCheckpoint
+from great_expectations.data_context.types.base import CheckpointValidationConfig
+
+if TYPE_CHECKING:
+    from great_expectations.checkpoint.configurator import ActionDict
+    from great_expectations.checkpoint.types.checkpoint_result import CheckpointResult
+    from great_expectations.data_context.data_context.abstract_data_context import (
+        AbstractDataContext,
+    )
+
+
+# TODO: to be renamed once we're good to move this out of experimental
+class ExperimentalCheckpoint:
+    def __init__(
+        self,
+        name: str,
+        data_context: AbstractDataContext,
+        action_list: Sequence[ActionDict],
+        validations: list[dict] | list[CheckpointValidationConfig],
+    ) -> None:
+        self._checkpoint = LegacyCheckpoint(
+            name=name,
+            data_context=data_context,
+            action_list=action_list,
+            validations=validations,
+        )
```

Comment:
@billdirks had the great idea of just making this a passthrough - we can leverage the functional Checkpoint behavior but pick and choose which parts of the API we want to expose in this class